### PR TITLE
refactor: code cleanup in Targets

### PR DESCRIPTION
**Pull request by rodolforg**
And maybe it helps performance, if new array size is equal to previous one, because resize() wouldn't perform anything in this case.

### CODE REVIEW

**Comment by rodolforg** [comment]
And macOS keeps randomly failing in app_layerduplicate :(
image

File: synfig-core/src/modules/mod_bmp/trgt_bmp.cpp [hunk 1]
```diff
@@ -205,8 +201,6 @@ bmp::end_frame()
 {
 	if(file)
 		fclose(file);
-	delete [] color_buffer;
-	color_buffer=0;
```
Comment (rodolforg) [inline synfig-core/src/modules/mod_bmp/trgt_bmp.cpp:209]: I'm not sure if we have to replace both lines with `color_buffer.clear()`

**Review by ice0 — DISMISSED** [review]

File: synfig-core/src/modules/mod_bmp/trgt_bmp.cpp (hunk 1 above)
Comment (rodolforg) [reply synfig-core/src/modules/mod_bmp/trgt_bmp.cpp:209]: Done

**Comment by rodolforg** [comment]
Rebased.

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**